docs/source/howto/codes.rst

```diff
@@ -4,41 +4,264 @@
+The first one is achieved by manipulating the |Folder| object that the method receives as an input, whereas the second one requires the construction of the |CalcInfo| object (which is then returned by the method) and the |CodeInfo| object (which will be included in the |CalcInfo| one, see line 18 in the following code snippet).
+
+.. code-block:: python
+   :linenos:
 
-In order to work with an external simulation code in AiiDA, you need a calculation input plugin (subclassing the :py:class:`~aiida.engine.CalcJob` class) and an output parser plugin (subclassing the :py:class:`~aiida.parsers.Parser` class):
+    def prepare_for_submission(self, folder):
 
-Before starting to write a plugin, check on the `aiida plugin registry <https://aiidateam.github.io/aiida-registry/>`_ whether a plugin for your code is already available.
+        input_x = self.inputs['x']
+        input_y = self.inputs['y']
+        input_code = self.inputs['code']
+        input_filename = self.inputs['metadata']['options']['input_filename']
+        output_filename = self.inputs['metadata']['options']['output_filename']
+
+        with folder.open(input_filename, 'w', encoding='utf8') as handle:
+            handle.write('echo $(( {} + {} ))\n'.format(input_x.value, input_y.value))
+
+        codeinfo = CodeInfo()
+        codeinfo.code_uuid = input_code.uuid
+        codeinfo.stdout_name = output_filename
+        codeinfo.cmdline_params = [input_filename]
+
+        calcinfo = CalcInfo()
+        calcinfo.codes_info = [codeinfo]
+        calcinfo.retrieve_list = [output_filename]
+
+        return calcinfo
+
+
+The first block of code (lines 3-7) is just the unloading of the information stored in the ``spec`` into local variables.
+Note that this information is not accessed via the ``spec.inputs`` anymore, but by ``self.inputs``: by the time this method is executed the specs will have become properties of the ``CalcJob`` and now should contain the actual inputs provided by the user.
+
+The input required by the addition "code" is just a `bashscript` line with the value of the input nodes replaced appropriately.
+This is being created on line 9, using the ``folder.open`` method to get a handle to the file and simply wirting in it.
+This directory represented by the |Folder| object (along with all the files created in it) will not only be copied to the remote machine for the code to be run there, but will also be stored in the local repository of the calculation node.
+
+Next in lines 12-15 we are creating and setting up the ``codeinfo = CodeInfo()`` object.
+The ``code_uuid`` being passed in line 13 is necessary for the engine to get the base information from the |Code| node (in which computer the code is placed, what is the location of the executable, etc).
```

Comment:
Somehow the vertical bars are showing up in the [formatted output here](https://36-77234579-gh.circle-artifacts.com/0/html/howto/codes.html).